src/Microsoft.DotNet.Build.Tasks.TargetFramework.Sdk/src/build/Microsoft.DotNet.Build.Tasks.TargetFramework.Sdk.props

```
@@ -11,10 +11,12 @@
     <DotNetBuildTasksTargetFrameworkSdkAssembly Condition="'$(DotNetBuildTasksTargetFrameworkSdkAssembly)' == '' AND '$(MSBuildRuntimeType)' == 'core'">..\tools\netcoreapp2.1\Microsoft.DotNet.Build.Tasks.TargetFramework.Sdk.dll</DotNetBuildTasksTargetFrameworkSdkAssembly>
     <DotNetBuildTasksTargetFrameworkSdkAssembly Condition="'$(DotNetBuildTasksTargetFrameworkSdkAssembly)' == '' AND '$(MSBuildRuntimeType)' != 'core'">..\tools\net472\Microsoft.DotNet.Build.Tasks.TargetFramework.Sdk.dll</DotNetBuildTasksTargetFrameworkSdkAssembly>
     <_OriginalTargetFramework>$(TargetFramework)</_OriginalTargetFramework>
+    <TargetFrameworkPattern><![CDATA[(?<v1>netstandard2.0|netstandard2.1|net461|net462|net47|netcoreapp2.0|netcoreapp2.1|netcoreapp3.1|netcoreapp3.0)(-[^;]+)]]></TargetFrameworkPattern>
```

This string makes me 😢.  Could we do this:
```suggestion
    <TargetFrameworkPattern><![CDATA[(?<v1>((netstandard|netcoreapp)[1-9\.]+)|(net[1-4][1-9\.]+))(-[^;]+)]]></TargetFrameworkPattern>
```

---

Even though I worked on the regex engine for about a year, I don't understand why we need a named capture group here.

---

>  Even though I worked on the regex engine for about a year, I don't understand why we need a named capture group here.

We need a named capture here because we are replacing tfm-tfp -> tfm, earlier we didnt need that  because we were removing all occurences of -tfp (which wont be the case anywmore)

---

right, that makes sense. but why does it need to be a named capture group and can't be a normal capture group which has implicit identifier ($1, $2, etc.)?

---

nice, i wasnt aware of that, i have always used the name capture. i changed it to normal capture.

---

one more question, why is the CDATA annotation necessary?

---

i removed that. 

> why is the CDATA annotation necessary?

it was earlier necessary to specify the ```<v1>``` notation